services/resource-coordinator/src/InfrastructureManager.js

```diff
@@ -47,34 +76,50 @@ class OIHInfrastructureManager extends InfrastructureManager {
         return baseUri.toString();
     }
 
-    _createFlowAmqpCredentials(flow) {
-        return this._createAmqpCredentials(flow);
-    }
+    async _ensureRabbitMqCredentialsForFlowNode(flow, node) {
+        const creds = await this._getRabbitMqCredentials(flow, node);
+        if (creds) {
+            this._logger.trace(creds, 'Found created credentials');
+            return creds;
+        }
 
-    async _createAmqpCredentials(flow) {
-        const username = flow.id;
+        const username = `${flow.id}_${node.id}`.toLowerCase().replace(/[^a-z0-9]/g, '');
         const password = uuid();
 
-        this._logger.trace('About to create RabbitMQ user');
+        this._logger.trace({username, password}, 'About to create RabbitMQ user');
+        // @todo: create node user
         await this._rabbitmqManagement.createFlowUser({
             username,
             password,
             flow
         });
-        this._logger.trace('Created RabbitMQ user');
+        this._logger.trace({username}, 'Created RabbitMQ user');
 
-        return {
-            username,
-            password
-        };
+        const newCreds = {username, password};
+        await this._saveRabbitMqCredentials(flow, node, newCreds);
+
+        return newCreds;
+    }
+
+    _getRabbitMqCredentials(flow, node) {
+        return this._credentialsStore.get(flow.id, node.id);
     }
 
-    _deleteFlowAmqpCredentials(flow) {
-        return this._deleteAmqpCredentials({username: flow.id});
+    _saveRabbitMqCredentials(flow, node, credentials) {
+        return this._credentialsStore.set(flow.id, node.id, credentials);
     }
 
-    _deleteAmqpCredentials(credentials) {
-        return this._rabbitmqManagement.deleteUser(credentials);
+    _deleteRabbitMqCredentials(flow, node) {
+        return this._credentialsStore.remove(flow.id, node.id);
+    }
+
+    async _deleteRabbitMqCredentialsForFlow(flow) {
+        const flowCredentials = await this._credentialsStore.getAllForFlow(flow.id);
```

Comment:
Generally here we have a problem. Scenario:
1. assume that several flows was started, all of them was assigned credentials, .... 
2. resource coordinator was restarted for some reason (fail, update, node restart, ....)
3. from now _credentialsStore is empty
4. stop  some flow
5. Consequences: as _credentialsStore is empty,  secrets won't be removed.

Reply:
@anton-kotenko created an issue to solve this problem: https://github.com/openintegrationhub/openintegrationhub/issues/259.